airflow/providers/yandex/secrets/secrets_manager.py

```diff
@@ -0,0 +1,209 @@
+
+    You need to specify id of yandexcloud connection to connect to Yandex Lockbox with.
+    Also, you need to specify the Yandex Cloud folder ID to search for Yandex Lockbox secrets in.
+
+    :param connection_id: Specifies the connection ID to connect to Yandex Lockbox with.
+        If set to None (null), requests will use default connection_id from YandexCloudBaseHook.
+        Default: "yandexcloud_default"
+    :param folder_id: Specifies the folder ID to search for Yandex Lockbox secrets in.
+        If set to None (null), requests will use the connection folder_id.
+    :param connections_prefix: Specifies the prefix of the secret to read to get Connections.
+        If set to None (null), requests for connections will not be sent to Yandex Lockbox.
+        Default: "airflow/connections"
+    :param variables_prefix: Specifies the prefix of the secret to read to get Variables.
+        If set to None (null), requests for variables will not be sent to Yandex Lockbox.
+        Default: "airflow/variables"
+    :param config_prefix: Specifies the prefix of the secret to read to get Configurations.
+        If set to None (null), requests for variables will not be sent to Yandex Lockbox.
+        Default: "airflow/config"
+    :param sep: Specifies the separator used to concatenate secret_prefix and secret_id.
+        Default: "/"
+    """
+
+    def __init__(
+        self,
+        connection_id: str | None = None,
+        folder_id: str | None = None,
+        connections_prefix: str | None = "airflow/connections",
+        variables_prefix: str | None = "airflow/variables",
+        config_prefix: str | None = "airflow/config",
+        sep: str = "/",
+    ):
+        super().__init__()
+
+        self.connection_id = connection_id
+        self.folder_id = folder_id
+        self.connections_prefix = connections_prefix.rstrip(sep) if connections_prefix is not None else None
+        self.variables_prefix = variables_prefix.rstrip(sep) if variables_prefix is not None else None
+        self.config_prefix = config_prefix.rstrip(sep) if config_prefix is not None else None
+        self.sep = sep
+
+    @property
+    def _client(self):
+        """Create a Yandex Cloud SDK client."""
+        return self._base_hook.client
+
+    @cached_property
+    def _base_hook(self) -> YandexCloudBaseHook:
+        return YandexCloudBaseHook(
+            yandex_conn_id=self.connection_id,
+            default_folder_id=self.folder_id,
+        )
+
+    def get_conn_value(self, conn_id: str) -> str | None:
+        """
+        Retrieve from Secrets Backend a string value representing the Connection object.
+
+        :param conn_id: connection id
+        :return: Connection Value
+        """
+        if self.connections_prefix is None:
+            return None
+
+        if conn_id == self.connection_id:
+            return None
+
+        if not self.connection_id and conn_id == YandexCloudBaseHook.default_conn_name:
```

Comment:
Not clear about this condition: `conn_id == YandexCloudBaseHook.default_conn_name`, thy it is required.

Reply:
It's similar to the previous one. It's needed because the secrets backend needs a connection to connect to Lockbox. It will cause a recursion issue if the secrets manager backend tries to answer a connection request from itself.

```
conn_id == self.connection_id
```

This happens when the user explicitly specifies the connection_id.

```
not self.connection_id and conn_id == YandexCloudBaseHook.default_conn_name.
```

This happens when the user does not specify a connection_id. In this case, the Yandex Cloud SDK will use `YandexCloudBaseHook.default_conn_name`.

Reply:
Resolve in favor of https://github.com/apache/airflow/pull/36449#discussion_r1456111043